### PR TITLE
Fix a few failures in post_fail_hook which make guests failed to be restored

### PR DESCRIPTION
**Pull request by Julie-CAO**
- Fix a few failures in post_fail_hook which blocks consequent tests running after sriov tests. 
  - publishing unit log failed due to no command history. In this case, [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/11163170#) , guests were not restored after sriov failure which stopped consequent tests.  
  - correct errors in uploading logs in post_fail_hook.
  - stop SLE test going into ALP environment. fg. http://10.67.129.102/tests/634#step/sriov_network_card_pci_passthrough/461
  - remove duplicate collecting virt logs from product tests as it has been uploaded in collect_host_and_guest_logs().
- remove Perl warnings in autoinst.txt, https://openqa.suse.de/tests/11181432/logfile?filename=autoinst-log.txt
```
Subroutine set_ssh_console_timeout_before_use redefined at sle/tests/virt_autotest/login_console.pm line 22.
```

Related ticket: https://progress.opensuse.org/issues/120750
Verification run: 
[storage_tests_continued_after_sriov_test_fail](https://openqa.suse.de/tests/11194705)
[logs_uploaded_correctly_in_post_fail_hook](http://10.67.129.102/tests/634)

### CODE REVIEW

**Comment by Julie-CAO** [comment]
@alice-suse @xguo @waynechen55 @nanzhg @tbaev @RoyCai7 @varunkojha  @tonyyuan1 Welcome review!

**Comment by alice-suse** [comment]
LGTM

File: lib/virt_autotest/utils.pm [hunk 1]
```diff
@@ -547,7 +547,8 @@ sub remove_additional_nic {
 
 sub collect_virt_system_logs {
     if (script_run("test -f /var/log/libvirt/*d.log") == 0) {
-        upload_logs("/var/log/libvirt/*d.log");
+        script_run('tar czvf /tmp/libvirt_daemons.tar.gz /var/log/libvirt/*d.log');
```
Comment (waynechen55) [inline lib/virt_autotest/utils.pm:550]: Default timeout enough for generate tar ball if large and many log files ?

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:550]: I just made a search that most of 'tar' runs without a timeout option. I think it is ok in most cases.
The function is used by MU tests and I am unable to tell or verify a clear result. Let MU testers handle it if timeout happens in the future.

File: lib/virt_feature_test_base.pm [hunk 2]
```diff
@@ -144,7 +144,8 @@ sub analyzeResult {
     #the correctness and effectivenees of entire JUnit log
     if ($status eq 'FAILED' && $self->{"fail_nums"} eq '0') {
         $self->{"fail_nums"} = '1';
-        my $uncheckpoint_failure = script_output("cat /root/commands_history | tail -3 | head -1");
+        my $uncheckpoint_failure = script_output("cat /root/commands_history | tail -3 | head -1", 60, proceed_on_failure => 1);
+        $uncheckpoint_failure = "echo 'NOT found the failure from commands history'" if $uncheckpoint_failure eq '';
```
[the remark this answers is not in the document]
Comment (Julie-CAO) [reply lib/virt_feature_test_base.pm:148]: ok, no problem.

**Comment by Julie-CAO** [comment]
PR is ready for merge. @alice-suse

**Pull request merged**